cirq/study/trial_result.py

```diff
@@ -66,15 +65,32 @@ def _big_endian_int(bits: Iterable[Any]) -> int:
     return result
 
 
+def _big_endian_bits(val: int, bit_count: int) -> List[bool]:
```

Comment:
I hate to reopen the endianness debate, but just curious what was the thinking to go with big-endian here? I think it at least merits duscussion, since we use little-endian bit order in internal code.

Reply:
I chose big endian to satisfy the following invariant:

```python
def invariant(a: np.ndarray, ks: Tuple[int, ...]):
    b = a.reshape(a.size)
    k = bits_to_int(ks)
    assert a[ks] == b[k]
```

If `bits_to_int` is a little endian conversion, the assertion fails. If it is big endian, the assertion succeeds.

In other words, we're big endian because numpy is big endian. Later indices in a numpy array have smaller strides in the same way that later big endian digits have smaller weights.

Reply:
> Later indices in a numpy array have smaller strides in the same way that later big endian digits have smaller weights.
This is true only for "C-contiguous" arrays; "Fortran-contiguous" arrays go the other way, which you could get by doing
```python
b = a.reshape(a.size, order='F')
```
in your example. So numpy supports both in a fairly straightforward way, but I guess it's good to stick with the default.

Reply:
You can also create similar invariants around merging indices for the result of np.kron.

If it comes down to it, we can add order='F' at some later date. For now this will do.